extension/app/background.ts

```diff
@@ -1,3 +1,173 @@
+import type {
+  Auth0AuthorizeResponse,
+  AuthBackgroundResponse,
+  AuthBackroundMessage,
+} from "./src/lib/auth";
+import {
+  AUTH0_AUDIENCE,
+  AUTH0_CLIENT_DOMAIN,
+  AUTH0_CLIENT_ID,
+} from "./src/lib/auth";
+import { generatePKCE } from "./src/lib/utils";
+
+/**
+ * Listener to open/close the side panel when the user clicks on the extension icon.
+ */
 chrome.runtime.onInstalled.addListener(() => {
   void chrome.sidePanel.setPanelBehavior({ openPanelOnActionClick: true });
 });
+
+/**
+ * Listener for messages sent from the react app to the background script.
+ * For now we use messages to authenticate the user.
+ */
+chrome.runtime.onMessage.addListener(
+  (
+    message: AuthBackroundMessage,
+    sender,
+    sendResponse: (
+      response: Auth0AuthorizeResponse | AuthBackgroundResponse
+    ) => void
+  ) => {
+    switch (message.type) {
+      case "AUTHENTICATE":
+        void authenticate(sendResponse);
+        return true; // Keep the message channel open for async response.
+
+      case "LOGOUT":
+        void logout(sendResponse);
+        return true; // Keep the message channel open.
+
+      default:
+        console.error(`Unknown message type: ${message.type}.`);
+    }
+  }
+);
+
+/**
+ * Authenticate the user using Auth0.
+ */
+const authenticate = async (
+  sendResponse: (auth: Auth0AuthorizeResponse | AuthBackgroundResponse) => void
+) => {
+  if (!AUTH0_CLIENT_ID || !AUTH0_CLIENT_DOMAIN) {
+    console.error("Auth0 client ID or domain is missing.");
+    return;
+  }
+
+  // First we call /authorize endpoint to get the authorization code (PKCE flow).
+  const redirectUrl = chrome.identity.getRedirectURL();
+  const { codeVerifier, codeChallenge } = await generatePKCE();
+  const options = {
```

Comment:
Looking a bit closer at Auth0 SDKs, why don't we use the SPA [SDK](https://auth0.com/docs/libraries/auth0-single-page-app-sdk)? You'll need to refresh the token silently anyway.

Reply:
Discussed IRL we tried at some point during the hackday and did not work but I think was the ReactSDK, in any case I think it's worth trying again indeed cause it would be better to use this. 
And if we cannot use it we should document it. 
Will explore this morning 👍🏻

Reply:
I tested the react SDK and _think_ it's not working cause auth it not launched by the background script. Will keep digging.

Reply:
Attempt here: https://github.com/dust-tt/dust/pull/8015

=> It's not working cause the SDK expects being in a browser and access things that are not available in an extension. 
=> I'm stopping the explo and will keep this version (applying all other feedback before merging of course)